### PR TITLE
Suggestion for questinfo requirement update

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: 

* **Server Mode**: 

* **Description of Pull Request**: 

## Previously
```
*questinfo <Quest ID>,<Icon>{,<Map Mark Color>{,<Job Class>}};
*setquestinfo_level <quest_id>,<min_level>,<max_level>;
*setquestinfo_req <quest_id>,<req_quest_id>,<state>{,<req_quest_id>,<state>,...};
*setquestinfo_job <quest_id>,<job_id>{,<job_id>...};
```
## Suggested
```
*questinfo <Icon>{,<Map Mark Color>{,"<condition>"}};
```
- `<Quest ID>` is removed, all the conditions are handled by the `<condition>` argument.
- The `<condition>` is using the `achievement_check_condition` (like in achievement_db) so any condition (in theory) can be used to display the icon.

I didn't found a better format than the `"<condition>"` to handle the variable which are being used in official script..

### CODE REVIEW

**Comment by JohnnyPlayy** [comment]
I love these changes that you've been doing, you're getting well organized and still making the scripts look better.

File: src/map/script.cpp [hunk 1]
```diff
@@ -19539,7 +19528,7 @@ BUILDIN_FUNC(showevent)
 {
 	TBL_PC *sd;
 	struct npc_data *nd = map_id2nd(st->oid);
-	int icon, color = 0;
+	int icon, color = QMARK_NONE;
```
Comment (aleos89) [inline src/map/script.cpp]:
```suggestion
	questinfo_types icon;
	e_questinfo_markcolor color = QMARK_NONE;
```

File: src/map/script.cpp [hunk 2]
```diff
@@ -19550,10 +19539,10 @@ BUILDIN_FUNC(showevent)
 	icon = script_getnum(st, 2);
```
Comment (aleos89) [inline src/map/script.cpp]:
```suggestion
	icon = static_cast<questinfo_types>(script_getnum(st, 2));
```

File: src/map/clif.cpp [hunk 3]
```diff
@@ -17051,9 +17051,12 @@ void clif_quest_update_status(struct map_session_data *sd, int quest_id, bool ac
 ///     1 = orange
 ///     2 = green
 ///     3 = purple
-void clif_quest_show_event(struct map_session_data *sd, struct block_list *bl, short effect, short color)
+void clif_quest_show_event(struct map_session_data *sd, struct block_list *bl, enum e_questinfo_types effect, enum e_questinfo_markcolor color)
```
Comment (secretdataz) [inline src/map/clif.cpp]: We need to stop putting `enum` or `struct` in front of parameter types

**Comment by aleos89** [comment]
Was there anything left on this?

**Comment by Atemo** [comment]
Unless someone has suggestion, it's ready to be merged. Nothing left on my side.

**Comment by officialwedevgames** [comment]
I dont know if this is bug or not.

questinfo QTYPE_QUEST2, QMARK_YELLOW, "checkquest(8641) == 1 && countitem(503) >= 1";

I tried this and still don't show the yellow color in the map and no icon.

this is the original code. if you didnt get yet the quest it will appear as green but if you already get it and you have the item required. Its should be yellow.

questinfo QTYPE_QUEST, QMARK_GREEN, "checkquest(8641) == -1";
questinfo QTYPE_QUEST2, QMARK_YELLOW, "isbegin_quest(8641) == 1 && countitem(503) >= 1";

**Comment by Atemo** [comment]
I don't see the issue after testing your 3 `questinfo`. For more support please ask on the forum.

**Pull request merged**